src/euphorie/client/notifications/base.py

```diff
@@ -103,6 +103,15 @@ def index(self):
             )
         )
 
+    def send_email(self):
+        if not self.account.email:
```

Comment:
This won't work for DTMB because the email address is not in the account there. On the other hand in plain euphorie this is unnecessary because the user ID is an email address, so every account has an email.

You probably need to use `webhelpers.get_user_email()`.

Reply:
IMO you should not even reach this point.
Results should be filter out earlier.

Reply:
@reinhardt tnx for the pointer. fixed.

@ale-rt that would involve two things:
a) Implementing this logic and the logging message in all of the different notification classes.
b) or restructuring the code so that the notification base class does that.

I think we can't or shouldn't filter out on the SQL query level - the ws_user_cache reponse JSON‌ field wouldn't allow that easily (not considering Postgres' JSON capabilities, which I‌ don't have experience with) and we would loose the logging feature.

I'm :-1: for a) and :+0: for b) - we're over budget and we would need a full test circle again.

Reply:
Yes, Alex is going to be furious if we do more than this quick fix.